src/binomial.jl

```diff
@@ -128,6 +128,11 @@ function StatsBase.confint(x::BinomialTest; level::Float64=0.95, tail=:both, met
     else
         throw(ArgumentError("tail=$(tail) is invalid"))
     end
+
+    # make sure we stay in [0, 1]
```

Comment:
Most CI methods already guarantee this property, it seems. Wouldn't it be better to directly fix the relevant `ci_...` methods below (AFAICT only Wald, Agresti-Coull, and Wilson (?) might be problematic)?

Reply:
> Wouldn't it be better to directly fix the relevant `ci_...` methods below

Why would that be better? They'll all need to be in the unit interval anyway so it seems more straightforward and future-proof to just guarantee it for everything.

Reply:
The motivation for the suggestion is two-fold: It avoids unnecessary (even though cheap) computations when they are not needed and it ensures that there are no surprisingly different results when calling `ci_wald` etc. directly.